src/features/broker-protection/actions/fill-form.js

```diff
@@ -4,36 +4,15 @@ import { ErrorResponse, SuccessResponse } from '../types.js'
 /**
  * @param {Record<string, any>} action
  * @param {Record<string, any>} userData
+ * @param {Document | HTMLElement} [root]
  * @return {import('../types.js').ActionResponse}
  */
-export function fillForm (action, userData) {
-    const form = getElement(document, action.selector)
+export function fillForm (action, userData, root = document) {
```

Comment:
Is the idea behind adding the root parameter that we could potentially do something similar to what you've done with extract (where you can pass a parent element)?